src/tic-tac-toe.cpp

```diff
@@ -10,23 +7,134 @@ using namespace std;
 
 class Player {
 public:
-    Player(string name, bool isCross)
-    : d_name(name),
-    d_isCross(isCross)
+    Player(string name)
+    : d_name(name)
     {    
     }
 
     string getName() {
         return d_name;
     }
 
-    bool getIsCross() {
-        return d_isCross;
+private:
+    string d_name;
+};
+
+class Field {
+public:
+    enum Status { 
+    e_WIN = 0, 
+    e_DRAW,
+    e_IN_PROGRESS, 
+    e_ERROR
+    };
+
+    Field (int m, int n, int k) 
+        : d_k(k),
+        d_cellsLeft(m * n)
+    {
+        for (int i = 0; i < m; i++) {
+            d_grid.push_back({});
+            for (int j = 0; j < n; j++) {
+                d_grid.back().push_back(nullptr);
+            }
+        }
+    }
+
+    Status makeMove(const Player& player, int x, int y) {
+        if ((x >= d_grid.size()) || (x < 0) ||
+            (y >= d_grid.back().size()) || (y < 0) || 
+            (d_grid[x][y] != nullptr)) {
+
+            return e_ERROR;
+        }
+        d_grid[x][y] = &player;
+        d_cellsLeft--;
+
+        return status(x, y);
+    }
+
+    void getGrid(const Player& player1, const Player& player2) const {
```

Comment:
You don't need to pass player into this method. Let's solve this problem in more general way, and assume that number of players can be in range 1 to infinity